### PR TITLE
Support .asciidoctorconfig.adoc at root of the workspace folder

### PR DESCRIPTION
**Pull request by apupier**
part of #380

### CODE REVIEW

File: src/features/asciidoctorConfig.ts
Comment (Mogztter) [inline src/features/asciidoctorConfig.ts:25]:
```suggestion
  return getConfigContent(workspaceFolder, '.asciidoctorconfig.adoc')
```

File: src/features/asciidoctorConfig.ts
Comment (Mogztter) [inline src/features/asciidoctorConfig.ts:20]: I might be wrong but it seems that the Intellij plugin will return the content of both files. So if you have:

**.asciidoctorconfig**
```
:name: Aurélien
:version: 1.2.3
```
**.asciidoctorconfig.adoc**
```
:name: Alexander
```

It will include the following contents:

```
:name: Aurélien
:version: 1.2.3


:name: Alexander


```

https://github.com/asciidoctor/asciidoctor-intellij-plugin/blob/64707c7632e1c5e6efc34100dc3cb2dd0bab4ea6/src/main/java/org/asciidoc/intellij/AsciiDocWrapper.java#L710-L745

So `.asciidoctorconfig.adoc` has higher precedence and both files are taken into account.

Comment (Mogztter) [reply src/features/asciidoctorConfig.ts:20]: @ahus1 is that correct?

Comment (ahus1) [reply src/features/asciidoctorConfig.ts:20]: @Mogztter - yes, you're right, this is the order it is being processed in. 

I doubt that a user should have both files in a folder, still it makes sense to have a consistent behavior across tools.

Thanks for paying extra attention here!

File: CHANGELOG.md
Comment (ahus1) [inline CHANGELOG.md:7]: In English text, there shouldn't be a blank before the comma.

```suggestion
- support `.asciidoctorconfig` and `.asciidoctorconfig.adoc` at root of the workspace by @apupier, @mogztter and @ahus1 (#380)
```

File: src/features/asciidoctorConfig.ts
Comment (apupier) [reply src/features/asciidoctorConfig.ts]: Thanks for pointing it out. I updated the Pull Request to have the same behavior

**Comment by apupier** [comment]
@Mogztter and @ahus1 Please do a second review. i incorporated the feedback. And sorry for pining instead of only requesting a review but GitHub is drunk, as soon as I ask for a review on one of you, it dismiss the ask for review on the other...

File: src/features/asciidoctorConfig.ts
Comment (Mogztter) [inline src/features/asciidoctorConfig.ts:24]: The Intellij IDEA plugin joins configuration files using two newlines:

https://github.com/asciidoctor/asciidoctor-intellij-plugin/blob/64707c7632e1c5e6efc34100dc3cb2dd0bab4ea6/src/main/java/org/asciidoc/intellij/AsciiDocWrapper.java#L727

File: src/features/asciidoctorConfig.ts
Comment (Mogztter) [inline src/features/asciidoctorConfig.ts:21]: To simplify the code we can do the following:

1. Create an array of configuration files
2.  Filter `undefined` values in the array
3. If the array is empty, return `undefined`
4. Otherwise, join the array using `\n\n`

Pseudo-code (untested)
```ts
const configContentFromAsciidoctorConfig = await getConfigContent(workspaceFolder, '.asciidoctorconfig')
const configContentFromAsciidoctorConfigDotAdoc = await getConfigContent(workspaceFolder, '.asciidoctorconfig.adoc')

const configContents = [
  configContentFromAsciidoctorConfig,
  configContentFromAsciidoctorConfigDotAdoc
].filter(config => config !== undefined)

if (configContents.length > 0) {
  return configContents.join(`\n\n`)
}
return undefined
```

File: src/features/asciidoctorConfig.ts
Comment (apupier) [reply src/features/asciidoctorConfig.ts:24]: sounds overkill as we already have two newlines at end of the grabbed content but well, better to stay as close as possible to IntelliJ

File: src/features/asciidoctorConfig.ts
Comment (apupier) [reply src/features/asciidoctorConfig.ts:21]: I find it clearer having the 4 cases in different if/else cases than using a loop but the end result is the same and higher chance that you will have to maintain it than me. I updated with your proposal.

**Comment by ahus1** [comment]
Hi, unfortunately I didn't find the time yet to look at this PR again; will try my best to do this later this week. Sorry for the delay. Feel free to proceed without my input, I don't want to block your efforts here.

**Review by Mogztter — APPROVED** [review]
It looks good, thanks!

**Pull request merged**
